File: Problems_and_Solutions/days-10 to 19/day-10/Java/Solution1.java
import java.util.Arrays;
class Solution1 {
    public static void quickSort(int a[], int l, int r){
        if(l < r){
            int pi = partition(a,l,r);
            quickSort(a,l,pi-1);
            quickSort(a,pi+1,r);
        }
    }
    
    public static int partition(int a[], int l, int r){
        int pivot = a[r];
        int i=l-1;
        for(int j=l; j<r; j++){
            if(a[j] < pivot){
                i++;
                int temp = a[i];
                a[i] = a[j];
                a[j] = temp;
            }
        }
        int temp = a[i+1];
        a[i+1] = a[r];
        a[r] = temp;
        return i+1;
    }

    public static void main(String[] args){
        int a[] = {4, 1, 3, 9, 7};
        quickSort(a,0,a.length-1);
        System.out.println(Arrays.toString(a));
    }
}

File: Problems_and_Solutions/days-10 to 19/day-10/cpp/solution 1.cpp
#include <bits/stdc++.h> 

int partition(vector<int>& arr,int low,int high)
{
    int pivot=arr[low];
    int j=low;
    int temp=0;
    for(int i=low+1;i<high;i++)
    {
        if(arr[i]<pivot)
        {
            j++;
            temp=arr[j];
            arr[j]=arr[i];
            arr[i]=temp;
        }

    }
    arr[low]=arr[j];
    arr[j]=pivot;
    return j;
}

void recursiveSort(vector<int>& arr,int low,int high){
    if(low<high)
    {
    int partitionIndex=partition(arr,low,high);
    recursiveSort(arr,low,partitionIndex);
    recursiveSort(arr,partitionIndex+1,high);
    }
}


vector<int> quickSort(vector<int> arr)
{
    // Write your code here.
    recursiveSort(arr,0,arr.size());
    return arr;
    
    
}

